awx_collection/plugins/lookup/tower_schedule_rrule.py

```diff
@@ -0,0 +1,235 @@
+      end_on:
+        description:
+          - How to end this schedule
+          - If this is not defined this schedule will never end
+          - If this is a positive integer this schedule will end after this number of occurances
+          - If this is a date in the format YYYY-MM-DD [HH:MM:SS] this schedule end after this date
+          - Used for all types except none
+        type: str
+      on_days:
+        description:
+          - The days to run this schedule on
+          - A comma seperated list which can contain values sunday, monday, tuesday, wednesday, thursday, friday
+          - Used for week type schedules
+      month_day_number:
+        description:
+          - The day of the month this schedule will run on (0-31)
+          - Used for month type schedules
+          - Can not be used with on_the parameter
+        type: int
+      on_the:
+        description:
+          - A description on when this schedule will run
+          - Two strings seperated by space
+          - First string is one of first, second, third, fourth, last
+          - Second string is one of sunday, monday, tuesday, wednesday, thursday, friday
+          - Used for month type schedules
+          - Can not be used with month_day_number parameters
+    extends_documentation_fragment: awx.awx.auth
+"""
+
+EXAMPLES = """
+    - name: Create a string for a schedule
+      debug:
+        msg: "{{ query('awx.awx.tower_schedule_rrule', 'none', start_date='1979-09-13 03:45:07') }}"
+"""
+
+RETURN = """
+_raw:
+  description:
+    - String in the rrule format
+  type: string
+"""
+
+from ansible.plugins.lookup import LookupBase
+from ansible.errors import AnsibleError
+from datetime import datetime
+import re
+
+missing_modules = []
+try:
+    import pytz
+except ImportError:
+    missing_modules.append('pytz')
+
+try:
+    from dateutil.rrule import rrule, MONTHLY, WEEKLY, DAILY, HOURLY, MINUTELY, MO, TU, WE, TH, FR, SA, SU
+except ImportError:
+    missing_modules.append('python.dateutil')
+
+if len(missing_modules) > 0:
+    raise AnsibleError('You are missing the modules {0}'.format(', '.join(missing_modules)))
+
+
+class LookupModule(LookupBase):
+    frequencies = {
+        'none': DAILY,
+        'minute': MINUTELY,
+        'hour': HOURLY,
+        'day': DAILY,
+        'week': WEEKLY,
+        'month': MONTHLY,
+    }
+
+    weekdays = {
+        'monday': MO,
+        'tuesday': TU,
+        'wednesday': WE,
+        'thursday': TH,
+        'friday': FR,
+        'saturday': SA,
+        'sunday': SU,
+    }
+
+    set_positions = {
+        'first': 1,
+        'second': 2,
+        'third': 3,
+        'fourth': 4,
+        'last': -1,
+    }
+
+    def parse_date_time(self, date_string):
+        try:
+            return datetime.strptime(date_string, '%Y-%m-%d %H:%M:%S')
+        except ValueError:
+            return datetime.strptime(date_string, '%Y-%m-%d')
+
+    def run(self, terms, variables=None, **kwargs):
```

Comment:
so is this `variables` an artifact of the general plugin interface for lookup plugins? And it's taken here just to not be used?

Reply:
Correct, that is just how I understand run needs to be declared.